torch/csrc/autograd/utils/python_arg_parsing.h

```diff
@@ -32,4 +32,35 @@ inline std::tuple<c10::optional<at::Device>, c10::optional<at::ScalarType>, bool
     );
   }
 }
+
+// Given a PyObject* returns an IValue that wraps the Python value based on
+// actual type.
+inline c10::IValue extract_pyobject(PyObject* obj) {
```

Comment:
Doesn't that already exist?

Reply:
@albanD I tried to find something equivalent, but didn't find any. If you are aware of anything to convert `PyObject` to `IValue`, please let me know, happy to reuse existing functionality.

Reply:
cc @suo do you have an idea? I feel like the jit already does that for sure.